boards/arm/v2m_musca/Kconfig.board

```diff
@@ -0,0 +1,14 @@
+#
+# Copyright (c) 2018 Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config BOARD_MUSCA_A
+	bool "ARM Cortex-M33 SMM on V2M-MUSCA"
+	depends on SOC_SERIES_MUSCA
+
+config BOARD_MUSCA_A_NONSECURE
+	bool "ARM Cortex-M33 SMM on V2M-MUSCA (Non-Secure)"
+	depends on SOC_SERIES_MUSCA
+	select ARM_NONSECURE_FIRMWARE
```

Comment:
The internal ARM option shall not be used directly, any more, since we, now, have top-level options for secure and non-secure firmware; `TRUSTED_EXECUTION_SECURE `and `TRUSTED_EXECUTION_NONSECURE`. These options can be selected in prj.conf of the user application. And the whole dependency framework for TrustZone-M and TEE is available, as well.

I don't think we need to define `BOARD_MUSCA_A_NONSECURE `option here.
If you really want to have a Non-Secure-only Board, you could just directly select `TRUSTED_EXECUTION_NONSECURE=y` in the Board defconfig.

Reply:
You could leave `BOARD_MUSCA_A_NONSECURE `as internal option (hidden) to implement your conditionals in Kconfig.defconfig, but it should be indirectly set by `TRUSTED_EXECUTION_NONSECURE`.

Reply:
I removed this whole part.  It is unnecessary.